code/modules/telesci/telesci_computer.dm

```diff
@@ -393,11 +393,14 @@ var/list/telesci_warnings = list(
 
 	if(href_list["eject_cell"])
 		if(cell)
-			usr.put_in_hands(cell)
+			if (usr.put_in_hands(cell))
+				usr.visible_message("<span class='notice'>[usr] removes the cell from \the [src].</span>", "<span class='notice'>You remove the cell from \the [src].</span>")
+			else 
+				visible_message("<span class='notice'>\The [src] beeps as its cell is removed.</span>")
+				cell.forceMove(get_turf(src))
 			cell.add_fingerprint(usr)
```

Comment:
you would still add a fingerprint for usr's that don't have hands. move this into `if (usr.put_in_hands(cell))`

Reply:
Fingerprints for things that don't have hands are stocked in "hidden_fingerprints" and are useful for admin investigation.
Add_fingerprint knows to add it to the correct list of fingerprints

Reply:
ah i see